### ISSUE TITLE
Hex symbol for the disassembly listing

### ISSUE THREAD

**Issue by lab313ru** [NationalSecurityAgency/ghidra#5575]
Is it possible to specify my own hex symbol for immediates in my own language? By default ghidra prints `0x` before a value.

**Comment by emteere** [comment]
Do you mean sleigh processor language?

**Comment by lab313ru** [comment]
Yes. For example, I would like to use $ for hex, like in motorola assemblers.

**Comment by emteere** [comment]
You can't change what is displayed in the sleigh language itself.   Obviously, you could put the "$" in sleigh before the language, as some languages put "#" in front of constants.  The 0x for the displayed value is put on by the format/display in the listing.  The default is to display in hex with the 0x, but convert will change it to display in other ways in the operand field and decompiler.

I'm not aware of anything like this. any way to turn that formating with the 0x off, or to change the value.

I suppose an option could be added for the operand field to display the 0x or not, and then you could embed the "$" in the language itself.  That would be very targetted to just the operand field.  It couldn't affect other places hex values are displayed or entered.

**Comment by lab313ru** [comment]
I see.
`#` is just a symbol to mark animmediate, then you should specify `$` to use hex.
Specifying `$` in sleigh is not good idea in case of displaying a decimal. So it would be great to do that like in IDA where you can specify your own symbol for hexes and it will be used only in case of hex-displaying this value. The same for binary 0b values.

**Comment by emteere** [comment]
Is this for exporting and feeding it into an assembler?
Is it a general preference that someone might choose on any processor because they like seeing "$" before hex values?

We have seen "h" used to display for hex values as well.

We've talked about display options like this before that would apply everywhere, even on entry of hex values.

**Comment by lab313ru** [comment]
That option is only available for modification during proc-module development. And yes, it's for asm exporting and for those who prefers to look at the real asm syntax.

**Comment by emteere** [comment]
Of note, the values coming out of Sleigh are all just values, so you can't specify say binary format for a particular instructions operand.

**Comment by lab313ru** [comment]
> Of note, the values coming out of Sleigh are all just values, so you can't specify say binary format for a particular instructions operand.

Sure. I mean ability to specify in a language config generally.
